tests/unit/test_solving.py

```diff
@@ -370,6 +370,8 @@ def quantum_number_problem_set(request) -> QNProblemSet:
     )
     problem_sets = stm.create_problem_sets()
     qn_problem_sets = [
-        p.to_qn_problem_set() for pl in problem_sets.values() for p in pl
+        p.to_qn_problem_set()
+        for strength in sorted(problem_sets)
+        for p in problem_sets[strength]
```

Comment:
@grayson-helmholz the instability of the test results was caused by the fact that the `problem_sets` is a `dict` (mapping of strength `float`s to lists). This makes the test stable and also enables asserting the exact number of solutions.